### ISSUE TITLE
Error on refresh of CRUD example page if navigate to Todo/Stocks 1st. Must clear localstorage/data. "Cannot read property 'entities' of undefined     at selectEntities"

### ISSUE THREAD

**Issue by chauey** [tomastrajan/angular-ngrx-material-starter#345]
## Minimal reproduction of the bug with instructions:
- navigate to Examples->Todos
- navigate to CRUD (do not add anything)
- refresh the page, it will have error in devtools console

## Expected behavior:
- expect to have no error when going to that page

## Other information:
Perhaps related to rehydrating app state from localstorage but not having code to handle missing collection data

## I would be willing to submit a PR to fix this issue:

[ ] Yes (Assistance is provided if you need help submitting a pull request)  
[ ] No

**Comment by tomastrajan** [comment]
@chauey thank you very much for your feedback we will have a look into this!

**Comment by timdeschryver** [comment]
The problem here is comes from `initStateFromLocalStorage`. 
The example state is getting partially loaded with `{todos: {...}}` and we're overriding the example state with this state from the local storage. Because we don't have a local storage state from our CRUD reducer, this CRUD state just disappears.

To solve this we'll have to spread the example state, instead of overriding the example state as a whole.
In other words the following line is wrong:

```
return { ...newState, ...LocalStorageService.loadInitialState() };
```

If you want you can give this a shot @chauey , otherwise I'll submit a PR.

**Comment by timdeschryver** [comment]
This is solved.
